packages/skill-evolution/lib/skill_evolution_common.py

```
@@ -138,19 +197,128 @@ def lessons_archive_path(project_dir: str, skill: str, config: dict | None = Non
+                if fcntl is not None:
+                    fcntl.flock(destination.fileno(), fcntl.LOCK_EX)
+                try:
+                    shutil.copyfileobj(source, destination)
+                    destination.flush()
+                finally:
+                    if fcntl is not None:
+                        fcntl.flock(destination.fileno(), fcntl.LOCK_UN)
+
+        os.rename(migrating_path, f"{legacy_path}.migrated.{claim_suffix}")
+    except Exception:
+        pass
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)